<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitXMLReporter -->
<testsuite ignored="0" hostname="73155.local" failures="0" tests="1" name="E2E Purchase Flow Test-0" time="20.362" errors="0" timestamp="2025-07-24T10:53:39 EDT">
  <testcase classname="org.latiffsyed.testUtils.AndroidBaseTest" name="@BeforeClass configureAppium" time="20.233">
    <failure type="io.appium.java_client.service.local.AppiumServerHasNotBeenStartedLocallyException" message="Appium HTTP server is not listening at http://127.0.0.1:4723/ after 20000 ms timeout. Consider increasing the server startup timeout value and check the server log for possible error messages occurrences.
Node.js executable path: /usr/local/bin/node
Arguments: [/usr/local/lib/node_modules/appium/build/lib/main.js, --port, 4723, --address, 127.0.0.1]
Server log: [38;5;112m[Appium][0m Welcome to Appium v2.19.0
[38;5;112m[Appium][0m Non-default server args:
[38;5;112m[Appium][0m { address: [32m&amp;apos;127.0.0.1&amp;apos;[39m }
[38;5;112m[Appium][0m The autodetected Appium home path: /Users/administrator/.appium
[38;5;112m[Appium][0m Attempting to load driver uiautomator2...
[38;5;112m[Appium][0m Attempting to load driver xcuitest...
[38;5;112m[Appium][0m Requiring driver at /Users/administrator/.appium/node_modules/appium-xcuitest-driver/build/index.js
[38;5;112m[Appium][0m Requiring driver at /Users/administrator/.appium/node_modules/appium-uiautomator2-driver/build/index.js
[38;5;112m[Appium][0m XCUITestDriver has been successfully loaded in 0.491s
[38;5;112m[Appium][0m AndroidUiautomator2Driver has been successfully loaded in 0.923s
[38;5;86m[HTTP][0m Could not start REST http interface listener. The requested port may already be in use. Please make sure there is no other instance of this server running already.
[38;5;112m[Appium][0m Could not configure Appium server. It&amp;apos;s possible that a driver or plugin tried to update the server and failed. Original error: listen EADDRINUSE: address already in use 127.0.0.1:4723
[38;5;112m[Appium][0m Error: listen EADDRINUSE: address already in use 127.0.0.1:4723
    at Server.setupListenHandle [as _listen2] (node:net:1939:16)
    at listenInCluster (node:net:1996:12)
    at node:net:2205:7
    at processTicksAndRejections (node:internal/process/task_queues:90:21)
">
      <![CDATA[io.appium.java_client.service.local.AppiumServerHasNotBeenStartedLocallyException: Appium HTTP server is not listening at http://127.0.0.1:4723/ after 20000 ms timeout. Consider increasing the server startup timeout value and check the server log for possible error messages occurrences.
Node.js executable path: /usr/local/bin/node
Arguments: [/usr/local/lib/node_modules/appium/build/lib/main.js, --port, 4723, --address, 127.0.0.1]
Server log: [38;5;112m[Appium][0m Welcome to Appium v2.19.0
[38;5;112m[Appium][0m Non-default server args:
[38;5;112m[Appium][0m { address: [32m'127.0.0.1'[39m }
[38;5;112m[Appium][0m The autodetected Appium home path: /Users/administrator/.appium
[38;5;112m[Appium][0m Attempting to load driver uiautomator2...
[38;5;112m[Appium][0m Attempting to load driver xcuitest...
[38;5;112m[Appium][0m Requiring driver at /Users/administrator/.appium/node_modules/appium-xcuitest-driver/build/index.js
[38;5;112m[Appium][0m Requiring driver at /Users/administrator/.appium/node_modules/appium-uiautomator2-driver/build/index.js
[38;5;112m[Appium][0m XCUITestDriver has been successfully loaded in 0.491s
[38;5;112m[Appium][0m AndroidUiautomator2Driver has been successfully loaded in 0.923s
[38;5;86m[HTTP][0m Could not start REST http interface listener. The requested port may already be in use. Please make sure there is no other instance of this server running already.
[38;5;112m[Appium][0m Could not configure Appium server. It's possible that a driver or plugin tried to update the server and failed. Original error: listen EADDRINUSE: address already in use 127.0.0.1:4723
[38;5;112m[Appium][0m Error: listen EADDRINUSE: address already in use 127.0.0.1:4723
    at Server.setupListenHandle [as _listen2] (node:net:1939:16)
    at listenInCluster (node:net:1996:12)
    at node:net:2205:7
    at processTicksAndRejections (node:internal/process/task_queues:90:21)

at io.appium.java_client.service.local.AppiumDriverLocalService.start(AppiumDriverLocalService.java:192)
at org.latiffsyed.core.utils.AppiumUtils.startAppiumServer(AppiumUtils.java:35)
at org.latiffsyed.testUtils.AndroidBaseTest.configureAppium(AndroidBaseTest.java:52)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: io.appium.java_client.service.local.AppiumServerAvailabilityChecker$ConnectionTimeout: ConnectionTimeout
at io.appium.java_client.service.local.AppiumServerAvailabilityChecker.waitUntilAvailable(AppiumServerAvailabilityChecker.java:72)
at io.appium.java_client.service.local.AppiumDriverLocalService.ping(AppiumDriverLocalService.java:144)
at io.appium.java_client.service.local.AppiumDriverLocalService.start(AppiumDriverLocalService.java:185)
... 20 more
Caused by: java.net.ConnectException: Connection refused
at java.base/sun.nio.ch.Net.pollConnect(Native Method)
at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:682)
at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:549)
at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:592)
at java.base/java.net.Socket.connect(Socket.java:751)
at java.base/sun.net.NetworkClient.doConnect(NetworkClient.java:178)
at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:531)
at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:636)
at java.base/sun.net.www.http.HttpClient.<init>(HttpClient.java:280)
at java.base/sun.net.www.http.HttpClient.New(HttpClient.java:386)
at java.base/sun.net.www.http.HttpClient.New(HttpClient.java:408)
at java.base/sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(HttpURLConnection.java:1319)
at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1252)
at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1138)
at java.base/sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:1067)
at io.appium.java_client.service.local.AppiumServerAvailabilityChecker.connectToUrl(AppiumServerAvailabilityChecker.java:79)
at io.appium.java_client.service.local.AppiumServerAvailabilityChecker.waitUntilAvailable(AppiumServerAvailabilityChecker.java:61)
... 22 more
... Removed 13 stack frames]]>
    </failure>
  </testcase> <!-- @BeforeClass configureAppium -->
  <testcase classname="org.latiffsyed.testUtils.AndroidBaseTest" name="@BeforeClass configureAppium" time="20.24">
    <failure type="io.appium.java_client.service.local.AppiumServerHasNotBeenStartedLocallyException" message="Appium HTTP server is not listening at http://127.0.0.1:4723/ after 20000 ms timeout. Consider increasing the server startup timeout value and check the server log for possible error messages occurrences.
Node.js executable path: /usr/local/bin/node
Arguments: [/usr/local/lib/node_modules/appium/build/lib/main.js, --port, 4723, --address, 127.0.0.1]
Server log: [38;5;112m[Appium][0m Welcome to Appium v2.19.0
[38;5;112m[Appium][0m Non-default server args:
[38;5;112m[Appium][0m { address: [32m&amp;apos;127.0.0.1&amp;apos;[39m }
[38;5;112m[Appium][0m The autodetected Appium home path: /Users/administrator/.appium
[38;5;112m[Appium][0m Attempting to load driver uiautomator2...
[38;5;112m[Appium][0m Attempting to load driver xcuitest...
[38;5;112m[Appium][0m Requiring driver at /Users/administrator/.appium/node_modules/appium-xcuitest-driver/build/index.js
[38;5;112m[Appium][0m Requiring driver at /Users/administrator/.appium/node_modules/appium-uiautomator2-driver/build/index.js
[38;5;112m[Appium][0m XCUITestDriver has been successfully loaded in 0.491s
[38;5;112m[Appium][0m AndroidUiautomator2Driver has been successfully loaded in 0.923s
[38;5;86m[HTTP][0m Could not start REST http interface listener. The requested port may already be in use. Please make sure there is no other instance of this server running already.
[38;5;112m[Appium][0m Could not configure Appium server. It&amp;apos;s possible that a driver or plugin tried to update the server and failed. Original error: listen EADDRINUSE: address already in use 127.0.0.1:4723
[38;5;112m[Appium][0m Error: listen EADDRINUSE: address already in use 127.0.0.1:4723
    at Server.setupListenHandle [as _listen2] (node:net:1939:16)
    at listenInCluster (node:net:1996:12)
    at node:net:2205:7
    at processTicksAndRejections (node:internal/process/task_queues:90:21)
">
      <![CDATA[io.appium.java_client.service.local.AppiumServerHasNotBeenStartedLocallyException: Appium HTTP server is not listening at http://127.0.0.1:4723/ after 20000 ms timeout. Consider increasing the server startup timeout value and check the server log for possible error messages occurrences.
Node.js executable path: /usr/local/bin/node
Arguments: [/usr/local/lib/node_modules/appium/build/lib/main.js, --port, 4723, --address, 127.0.0.1]
Server log: [38;5;112m[Appium][0m Welcome to Appium v2.19.0
[38;5;112m[Appium][0m Non-default server args:
[38;5;112m[Appium][0m { address: [32m'127.0.0.1'[39m }
[38;5;112m[Appium][0m The autodetected Appium home path: /Users/administrator/.appium
[38;5;112m[Appium][0m Attempting to load driver uiautomator2...
[38;5;112m[Appium][0m Attempting to load driver xcuitest...
[38;5;112m[Appium][0m Requiring driver at /Users/administrator/.appium/node_modules/appium-xcuitest-driver/build/index.js
[38;5;112m[Appium][0m Requiring driver at /Users/administrator/.appium/node_modules/appium-uiautomator2-driver/build/index.js
[38;5;112m[Appium][0m XCUITestDriver has been successfully loaded in 0.491s
[38;5;112m[Appium][0m AndroidUiautomator2Driver has been successfully loaded in 0.923s
[38;5;86m[HTTP][0m Could not start REST http interface listener. The requested port may already be in use. Please make sure there is no other instance of this server running already.
[38;5;112m[Appium][0m Could not configure Appium server. It's possible that a driver or plugin tried to update the server and failed. Original error: listen EADDRINUSE: address already in use 127.0.0.1:4723
[38;5;112m[Appium][0m Error: listen EADDRINUSE: address already in use 127.0.0.1:4723
    at Server.setupListenHandle [as _listen2] (node:net:1939:16)
    at listenInCluster (node:net:1996:12)
    at node:net:2205:7
    at processTicksAndRejections (node:internal/process/task_queues:90:21)

at io.appium.java_client.service.local.AppiumDriverLocalService.start(AppiumDriverLocalService.java:192)
at org.latiffsyed.core.utils.AppiumUtils.startAppiumServer(AppiumUtils.java:35)
at org.latiffsyed.testUtils.AndroidBaseTest.configureAppium(AndroidBaseTest.java:52)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: io.appium.java_client.service.local.AppiumServerAvailabilityChecker$ConnectionTimeout: ConnectionTimeout
at io.appium.java_client.service.local.AppiumServerAvailabilityChecker.waitUntilAvailable(AppiumServerAvailabilityChecker.java:72)
at io.appium.java_client.service.local.AppiumDriverLocalService.ping(AppiumDriverLocalService.java:144)
at io.appium.java_client.service.local.AppiumDriverLocalService.start(AppiumDriverLocalService.java:185)
... 20 more
Caused by: java.net.ConnectException: Connection refused
at java.base/sun.nio.ch.Net.pollConnect(Native Method)
at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:682)
at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:549)
at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:592)
at java.base/java.net.Socket.connect(Socket.java:751)
at java.base/sun.net.NetworkClient.doConnect(NetworkClient.java:178)
at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:531)
at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:636)
at java.base/sun.net.www.http.HttpClient.<init>(HttpClient.java:280)
at java.base/sun.net.www.http.HttpClient.New(HttpClient.java:386)
at java.base/sun.net.www.http.HttpClient.New(HttpClient.java:408)
at java.base/sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(HttpURLConnection.java:1319)
at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1252)
at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1138)
at java.base/sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:1067)
at io.appium.java_client.service.local.AppiumServerAvailabilityChecker.connectToUrl(AppiumServerAvailabilityChecker.java:79)
at io.appium.java_client.service.local.AppiumServerAvailabilityChecker.waitUntilAvailable(AppiumServerAvailabilityChecker.java:61)
... 22 more
... Removed 13 stack frames]]>
    </failure>
  </testcase> <!-- @BeforeClass configureAppium -->
  <testcase classname="org.latiffsyed.tests.android.PurchaseOrderFlowTest" name="@BeforeMethod preSetup" time="0.008">
    <failure type="java.lang.NullPointerException" message="Cannot invoke &amp;quot;org.latiffsyed.pageobjects.android.FormPage.setActivity()&amp;quot; because &amp;quot;this.formPage&amp;quot; is null">
      <![CDATA[java.lang.NullPointerException: Cannot invoke "org.latiffsyed.pageobjects.android.FormPage.setActivity()" because "this.formPage" is null
at org.latiffsyed.tests.android.PurchaseOrderFlowTest.preSetup(PurchaseOrderFlowTest.java:66)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1583)
... Removed 19 stack frames]]>
    </failure>
  </testcase> <!-- @BeforeMethod preSetup -->
  <testcase classname="org.latiffsyed.tests.android.PurchaseOrderFlowTest" name="@BeforeMethod preSetup" time="0.008">
    <failure type="java.lang.NullPointerException" message="Cannot invoke &amp;quot;org.latiffsyed.pageobjects.android.FormPage.setActivity()&amp;quot; because &amp;quot;this.formPage&amp;quot; is null">
      <![CDATA[java.lang.NullPointerException: Cannot invoke "org.latiffsyed.pageobjects.android.FormPage.setActivity()" because "this.formPage" is null
at org.latiffsyed.tests.android.PurchaseOrderFlowTest.preSetup(PurchaseOrderFlowTest.java:66)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1583)
... Removed 19 stack frames]]>
    </failure>
  </testcase> <!-- @BeforeMethod preSetup -->
  <testcase classname="org.latiffsyed.testUtils.AndroidBaseTest" name="@AfterClass tearDown" time="0.009">
    <failure type="java.lang.NullPointerException" message="Cannot invoke &amp;quot;io.appium.java_client.android.AndroidDriver.quit()&amp;quot; because &amp;quot;this.driver&amp;quot; is null">
      <![CDATA[java.lang.NullPointerException: Cannot invoke "io.appium.java_client.android.AndroidDriver.quit()" because "this.driver" is null
at org.latiffsyed.testUtils.AndroidBaseTest.tearDown(AndroidBaseTest.java:72)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1583)
... Removed 14 stack frames]]>
    </failure>
  </testcase> <!-- @AfterClass tearDown -->
  <testcase classname="org.latiffsyed.testUtils.AndroidBaseTest" name="@AfterClass tearDown" time="0.009">
    <failure type="java.lang.NullPointerException" message="Cannot invoke &amp;quot;io.appium.java_client.android.AndroidDriver.quit()&amp;quot; because &amp;quot;this.driver&amp;quot; is null">
      <![CDATA[java.lang.NullPointerException: Cannot invoke "io.appium.java_client.android.AndroidDriver.quit()" because "this.driver" is null
at org.latiffsyed.testUtils.AndroidBaseTest.tearDown(AndroidBaseTest.java:72)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1583)
... Removed 14 stack frames]]>
    </failure>
  </testcase> <!-- @AfterClass tearDown -->
  <testcase classname="org.latiffsyed.tests.android.PurchaseOrderFlowTest" name="EndToEndOrderFlowTest" time="0.0">
    <skipped/>
  </testcase> <!-- EndToEndOrderFlowTest -->
</testsuite> <!-- E2E Purchase Flow Test-0 -->
